FileWriteFlags.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace FileWriteFlag
+{
+	// Default values are Do not overwrite files, append to existing files, and write in binary format.
+	enum FileWriteFlag : unsigned int
+	{
+		// SET:     Allow overwritting the file if it exists (mutually exclusive from Append).
+		// NOT SET: Do not allow overwritting an exisitng file.
+		Overwrite = 0b0000'0001,
+
+		// SET:     Do not allow a new file to be created (Mutually exclusive from Overwrite).
+		// NOT SET: Allow new file creation. Overwrite must also be set.
+		DoNotCreate = 0b0000'0010,
```

Comment:
A `Create` flag might be a little more straightforward. A write stream can default to having this flag set. A read stream would not.

Reply:
Maybe this is a dumb question, but how do I setup the program to have a flag default setting to ON? The reason I chose DoNotCreate is because the default value of 0 would then allow creating a new file. 

These flags are only designed for manipulating a File during Write situations.

Reply:
In the method that opens the file, you can set a default parameter value for the flags parameter. I realize the bit sensibleness of what you've done. I've done it plenty of times myself, sometimes with worse names. Though, I do find negatively named bit flags tend to be confusing. What about `OpenExisting`?